src/index.js

```diff
@@ -83,11 +84,13 @@ client.commands = new Collection();
 
 process.on('uncaughtException', (error) => {
   handleCriticalError(error);
+  registerGlobalErrorHandlers(client);
 });
 
 process.on('unhandledRejection', (reason, promise) => {
   console.error('Unhandled Rejection at:', promise);
   handleCriticalError(reason);
+  registerGlobalErrorHandlers(client);
```

Comment:
Here you're calling `registerGlobalErrorHandlers`, which starts two listeners, so each time it fails it will declare more listeners, it's weird, please take a look